graphene_sqlalchemy/types.py

```diff
@@ -21,70 +24,91 @@
 from .utils import get_query, is_mapped_class, is_mapped_instance
 
 
+class ORMField(OrderedType):
+    def __init__(
+        self,
+        type=None,
+        prop_name=None,
+        description=None,
+        deprecation_reason=None,
+        required=None,
+        _creation_counter=None,
+        **field_kwargs
+    ):
+        super(ORMField, self).__init__(_creation_counter=_creation_counter)
+        # The is only useful for documentation and auto-completion
+        common_kwargs = {
+           'type': type,
+           'prop_name': prop_name,
+           'description': description,
+           'deprecation_reason': deprecation_reason,
+           'required': required,
+        }
+        common_kwargs = {kwarg: value for kwarg, value in common_kwargs.items() if value is not None}
+        self.kwargs = field_kwargs
+        self.kwargs.update(common_kwargs)
+
+
 def construct_fields(
```

Comment:
I think this could benefit from—at the least—a docstring describing the overall flow and the intent. Even something that outlines the logic:

- gather all relevant attributes from the model
- gather all custom ORM fields defined on the type
- merge in overrides and build up all the fields

This logic is pretty dense and I think docstrings and inline comments would be beneficial to future readers.